AdvancedDLSupport/NativeLibraryBase.cs

```diff
@@ -106,6 +110,22 @@ protected void ThrowIfDisposed()
             }
         }
 
+        /// <summary>
+        /// Adds a delegate to keep it's lifetime till this library gets disposed.
+        /// </summary>
+        /// <param name="del">The delegate to keep alive.</param>
+        /// <returns>Returns the marshaled pointer.</returns>
+        protected IntPtr AddLifetimeDelegate(Delegate del)
+        {
+            if (del == null)
```

Comment:
Remove check and replace with a [NotNull] annotation instead. If you want to keep the check, throw an exception instead of returning null.

Reply:
This is done, as Marshal.GetFunctionPointerForDelegate does not accept null. And I need IntPtr.Zero as a marshalled value, when I call AddLifetimeDelegate. If I would not do it here I would need to implement that in IL ad hoc, which I wanted to prevent, I could still do it there, but I don't think it makes it better.
(null as delegate is btw. often a valid callback)

Reply:
Gotcha. A comment explaining this would be nice.